lib/src/kbaserelationengine/search/ElasticIndexingStorage.java

```diff
@@ -1064,18 +1192,36 @@ public FoundHits searchObjects(String objectType, MatchFilter matchFilter,
         if (accessGroupIds.isEmpty()) {
             return null;
         }
-        return createAccessMustBlock(accessGroupIds, accessFilter.withAllHistory);
+        return createAccessMustBlock(accessGroupIds, accessFilter.withAllHistory,
+                accessFilter.withPublic);
     }
     
     @SuppressWarnings("serial")
     private Map<String, Object> createAccessMustBlock(Set<Integer> accessGroupIds, 
-            boolean withAllHistory) {
+            boolean withAllHistory, boolean withPublic) {
+        List<Object> should = new ArrayList<>();
         String groupListProp = withAllHistory ? "groups" : "lastin";
         Map<String, Object> match = new LinkedHashMap<String, Object>() {{
             put(groupListProp, accessGroupIds);
         }};
-        Map<String, Object> query = new LinkedHashMap<String, Object>() {{
+        should.add(new LinkedHashMap<String, Object>() {{
             put("terms", match);
+        }});
+        if (withPublic) {
+            // Case of public workspaces containing DataPalette referencing to given object 
+            Map<String, Object> exists = new LinkedHashMap<String, Object>() {{
+                put("field", "extpub");
+            }};
+            should.add(new LinkedHashMap<String, Object>() {{
+                put("exists", exists);
```

Comment:
Need an explanation of what's going on here - looks like it's just checking for the existence of the pub ids list? Isn't that always going to exist?

Reply:
No, that's an essential parameter that you want to check: how many public workspaces (external comparing to home workspace of given object) have DataPalettes referencing given object (and version). If this number is 0 then object+version is not visible as public through DataPalettes. If it's >0 then it's visible.

Reply:
I see, the exists query isn't just that the field exists, it's that it exists and contains at least one non-null entry, so an empty list will be false